examples/lines_bars_and_markers/categorical_variables.py

```diff
@@ -0,0 +1,25 @@
+"""
+==============================
+Plotting categorical variables
+==============================
+
+How to use categorical variables in matplotlib.
+
+Many times you want to create a plot that uses categorical variables
+in Matplotlib. For example, your data may naturally fall into
+several "bins" and you're interested in summarizing the data per
+bin. Matplotlib allows you to pass categorical variables directly to
+many plotting functions.
+"""
+import matplotlib.pyplot as plt
+
+names = ['group_a', 'group_b', 'group_c']
```

Comment:
I'm in favor of more concrete examples. Here's some I did...and yes should have just added to matplotlib:
```python
data = {'apples':10, 'oranges':15, 'lemons':5, 'limes':20}
```
Also:
```python
cat = ["bored", "happy", "bored", "bored", "happy", "bored"]
dog = ["happy", "happy", "happy", "happy", "bored", "happy"]
time = ["combing", "drinking","feeding", "napping", "playing", "washing"]

fig = plt.figure()
ax = fig.add_subplot(1,1,1)
ax.plot(time, dog, label="dog")
ax.plot(time, cat, label="cat")
ax.legend()
```